CONTRIBUTING.md

```diff
@@ -470,8 +470,33 @@ doumentation are allowed to be extended with additional methods.
 
 > Warning: methods may be added to this interface in minor releases.
 
+These interfaces are defined by the OpenTelemetry specification and will be
+updated as the specification evolves.
+
 Otherwise, stable interfaces MUST NOT be modified.
 
+#### How to Change Specification Interfaces
+
+When an API change must be made, we will update the SDK with the new method one
+release before the API change. This will allow the SDK one version before the
+API change to work seemlessly with the new API.
+
+If an incompatible version of the SDK is used with the new API the application
```

Comment:
As has been noted before (e.g. in #3277), this will cause some pain on downstream libraries implementing these interfaces. This may be an acceptable tradeoff (reading through #3920 it certainly seems hard to do this in a painless way for end-users), but I think there should be *some* help from the OTel Go project on how to account for changes to these interfaces in terms of:
1. ensuring these changes are publicly documented in a way that makes it easy for downstream libraries to account for this changes (and if it is just hard for this information to reach the appropriate people, consider if waiting a single version is enough time for downstream users)
2. providing guidance for downstream libraries on how to support multiple versions of the interface (e.g. do we want downstream libraries to have separate implementations for each version of the interfaces as suggested on https://github.com/open-telemetry/opentelemetry-go/issues/3277#issuecomment-1308053982?)

I have not participated in this decision and I am an outsider to OTel Go so I don't know what the proper answer to these questions is, but it feels like before this wording is set in stone they should at least be considered.

Reply:
For metrics API the possibilities are documented here: https://pkg.go.dev/go.opentelemetry.io/otel/metric#hdr-API_Implementations

I think (not validated) that it could be possible to refactor the traces API in a similar way (by introducing `go.opentelemetry.io/otel/trace/embedded`) without introducing breaking changes. However, this would be rather a separate issue.

Reply:
Alright, filed #4160 for this, we can mark this thread as solved. Thanks for following up!